video/opengl.h

```diff
@@ -46,16 +46,14 @@ class OpenGL : public Input
 
     virtual bool readFrame(cv::Mat &outFrame) override
     {
-        // If outFrame is not of the expected size or format, recreate it
-        if (outFrame.cols != m_ReadWidth || outFrame.rows != m_ReadHeight || outFrame.type() != CV_8UC3) {
-            outFrame.create(m_ReadHeight, m_ReadWidth, CV_8UC3);
-        }
+        // Make sure frame is of right size and type
+        outFrame.create(m_ReadWidth, m_ReadHeight, CV_8UC3);
```

Comment:
I was initially paranoid/suspicious of this, but it's clear from the docs that this is a good idea as create has exactly this behaviour :smile:

Reply:
Yeah, I only just discovered this :-)